### PR TITLE
Add ADR for one database per state decision

### LINKED ISSUE #2408: Write ADR about per-state database decision
The vendor selection process showed us that vendors will likely propose changes in how we store our data. In making these decisions, It will be beneficial for the vendor to clearly understand why we originally decided to have a different database for each state (each state as a "tenant" of the NAC platform), and how the PPRL pivot changed those considerations.

### PR DESCRIPTION
**Pull request by echappen**
Expanded on @kingcomma's [original language ](https://gsa-tts.slack.com/archives/G01PJJS2JM8/p1638804814002300?thread_ts=1638206036.000700&cid=G01PJJS2JM8)

Closes #2408

### CODE REVIEW

File: docs/adr/0022-have-one-database-per-state.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+We believe the benefits of this approach largely dissolved once we pivoted to our [PPRL approach](../pprl.md), where deidentified PII from states is stored in the NAC and matches are performed directly against this deidentified data.
+
+We understand that having a separate NAC Postgresql database for each state now brings added complexity to the project without the original intended benefits. This opens up new possibilities for how to store this data more effectively. In deciding what will be best going forward, we believe the following considerations are still important:
+
+1. Using the principle of least privilege: states should have no access to other states' databases under any circumstances
+1. Authorization: states should not be able to modify their data once they've submitted it to the NAC. (In the current system, they can replace their data, but not modify or delete it.) There also may be no reason for states to read their own data once it's submitted.
+1. Security: how would the new system change the threat model to the data?
+1. Scalability: the future data store may contain hundreds of millions of records at a time
```
Comment (kingcomma) [inline docs/adr/0022-have-one-database-per-state.md:24]:
```suggestion
1. Scalability: the future data store will contain at least tens of millions of records at a time
```

File: docs/adr/0022-have-one-database-per-state.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+1. Performance: how can data be stored to make realtime matching faster?
```
Comment (kingcomma) [inline docs/adr/0022-have-one-database-per-state.md:25]:
```suggestion
1. Performance: how can data be stored to make near real-time matching faster?
```

File: docs/adr/0022-have-one-database-per-state.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+1. That it's a well-supported solution within the Microsoft/Azure ecosystem
```
Comment (kingcomma) [inline docs/adr/0022-have-one-database-per-state.md:26]:
```suggestion
1. Well established support within the Microsoft/Azure ecosystem
```

File: docs/adr/0022-have-one-database-per-state.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+1. Ease of use for the [Bulk Upload API](../openapi/generated/bulk-api/openapi.md)
+
+## Decision
+
+We will keep our current database structure while remaining open to possibilities for change. Making these changes will be weighed against other competing project priorities.
+
+## Consequences
+
+* We will need to support and maintain a separate Postgresql database for each state and ensure schema consistency across all of them.
```
Comment (kingcomma) [inline docs/adr/0022-have-one-database-per-state.md:35]:
```suggestion
* We will need to support and maintain a separate PostgreSQL database for each state and ensure schema consistency across all of them.
```

**Review by kingcomma — COMMENTED** [review]
Thanks for pulling this together! I added some suggestions, mostly cribbing from [a previous version of the project readme](https://github.com/18F/piipan/blob/352f1fbd8e2be984eb236fa581efbffa66c13a2a/README.md).

**Comment by echappen** [comment]
@kingcomma Thanks for the word-smithing! Added your changes.

**Review by kingcomma — APPROVED** [review]
👍 Cool — LGTM!

**Pull request merged**